### ISSUE TITLE
push_wait does not break, if the user presses decline on the smartphone.

### ISSUE THREAD

**Issue by cornelinux** [privacyidea/privacyidea#2865]
The push_wait only exits the waiting loop, if the challenge was answered successfully, but not if the authentication request was declined on the smartphone:

https://github.com/privacyidea/privacyidea/blob/master/privacyidea/lib/tokens/pushtoken.py#L916

(see https://community.privacyidea.org/t/connection-reject-does-not-work-as-expected-when-running-freeradius-as-a-service/2094/6?u=cornelinux)

We need to check, if the user presses decline on the smartphone, if this is only a smartphone internal functionality or if the challenge on the server gets modified/deleted. If so, we need to extend the if-statement in line 916, so that the loop also breaks if the user presses decline.

(I am not sure, if the decline button really communicates to the server, how the smartphone app is designed)

**Comment by nilsbehlen** [comment]
No the decline button does not communicate with privacyIDEA. I do not know if there is an endpoint that allowes to drop challenges? If there is, we could implement it.

Currently, pressing decline only removes the authentication request from the smartphone.

**Comment by cornelinux** [comment]
You are right. We do not have an endpoint, that could be used *without* a service account.
Actually *answering the challenge* works via `/validate/check` and the `nonce`. (see https://github.com/privacyidea/privacyidea/wiki/concept%3A-PushToken#smartphone-response)

There is actually no way to *decline* or *drop* the challenge. However, this should probably also be done via `/validate/check`, too.

Maybe by sending an additional parameter like `decline`, to still indicate, that this is a valid response from the smartphon:
~~~~http
POST https://privacyideaserver/validate/check
     nonce=<nonce_from_request>
     serial=<serial>
     signature=<signature>
     decline=1
~~~~

**Comment by nilsbehlen** [comment]
Sounds good to me.

**Comment by cornelinux** [comment]
Hi @nilsbehlen 
how do you like this? https://github.com/privacyidea/privacyidea/wiki/concept:-PushToken#decline

**Comment by nilsbehlen** [comment]
i like it, please add it to the next version of the server =)
